apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/AbstractCleanupJob.java

```diff
@@ -0,0 +1,155 @@
+package gov.cms.bfd.pipeline.rda.grpc;
+
+import gov.cms.bfd.pipeline.sharedutils.TransactionManager;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.Query;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.slf4j.Logger;
+
+/**
+ * An abstract class that encapsulates the common code for executing a job to clean up old
+ * pre-adjudicated claims from the RDA pipeline tables.
+ */
+@Getter
+@AllArgsConstructor
+public abstract class AbstractCleanupJob implements CleanupJob {
+
+  /** maximum age of claims in days from current date. */
+  private static final int OLDEST_CLAIM_AGE_IN_DAYS = 60;
+
+  /** template for delete query. */
+  private static final String DELETE_QUERY_TEMPLATE =
+      "delete from %s t where t.%s in ("
+          + "  select %s "
+          + "  from %s "
+          + "  where last_updated < '%s' "
+          + "  order by last_updated "
+          + "  limit %d"
+          + ")";
```

Comment:
Although @mackec1's changes where I saw a parameterized query recently are still in-flight under #2177, I think I like how that template reads a little better, as opposed to string interpolation. The subsequent call to `createNativeQuery` doesn't read so great, in my opinion. Would it be possible to define the parameters in-line here? E.g. from #2177 
```java
/* ..snipped.. */
  public static final String FIND_BENE_ID_FROM_IDENTIFIER_SQL =
      "SELECT * FROM find_beneficiary(:searchIdType, :searchIdValue)";
/* ..snipped ..*/
          entityManager
              .createNativeQuery(FIND_BENE_ID_FROM_IDENTIFIER_SQL)
              .setParameter("searchIdType", searchType)
              .setParameter("searchIdValue", searchValue)
              .getResultList();
/* ..snipped.. */
```

Reply:
That would work if we were just substituting the query parameters but won't work with table names.  I can do something similar using StringSubstitutor from the apache commons text lib, like this:
```java
  private static final String DELETE_QUERY_TEMPLATE =
      "delete from ${tableName} t where t.${parentTableKey} in ("
          + "  select ${parentTableKey} "
          + "  from ${parentTableName} "
          + "  where last_updated < '${cutoffDate}' "
          + "  order by last_updated "
          + "  limit ${limit}"
          + ")";
```
Otherwise I can just create out the query strings for each table.

Reply:
@rcrupp Yes! I think the `StringSubstitutor` strategy is simple enough and achieves what I was hoping for–I just saw the current implementation with it's positional arguments and was wanting for a little extra help while I was switching back and forth between the template definition and its usage. I think this avoids that kind of situation.